std/range/package.d

```diff
@@ -4416,6 +4416,33 @@ body
     assert(approxEqual(rf, [0.0, 0.1, 0.2, 0.3, 0.4]));
 }
 
+///
+unittest
+{
+    import std.stdio;
+
+    /* The following groups all produce the same output of:
+     * 0 1 2 3 4
+     */
+
+    foreach (i; 0..5)
+        writef("%s ", i);
+    writeln();
+
+
+    import std.range : iota;
+    foreach (i; iota(0, 5))
+        writef("%s ", i);
+    writeln();
+
+    writefln("%(%s %|%)", iota(0, 5));
+
+    import std.algorithm : map, copy;
+    import std.format;
+    iota(0, 5).map!(i => format("%s ", i)).copy(stdout.lockingTextWriter());
```

Comment:
I don't agree it is better, because it still allocates memory and is significantly more complex. Nobody is going to look at that and think it's better than the loop version. Also, I do not want to fix the trailing whitespace because the other examples do not fix it either. The code snippets need to do the same thing.